calico_kubernetes/calico_kubernetes.py

```diff
@@ -568,120 +574,53 @@ def _api_root_secure(self):
-
-        :param pod: pod info to parse
-        :type pod: dict()
-        :return:
-        """
-        try:
-            profile = self._datastore_client.get_profile(self.profile_name)
-        except:
-            logger.exception("ERROR: Could not apply rules. Profile not "
-                             "found: %s, exiting", self.profile_name)
-            sys.exit(1)
-
-        inbound_rules, outbound_rules = self._generate_rules(pod)
-
-        logger.info("Removing Default Rules")
-
-        # Remove default rules. Assumes 2 inbound, 1 outbound.
-        try:
-            self.calicoctl('profile', self.profile_name, 'rule', 'remove',
-                           'inbound', '--at=2')
-            self.calicoctl('profile', self.profile_name, 'rule', 'remove',
-                           'inbound', '--at=1')
-            self.calicoctl('profile', self.profile_name, 'rule', 'remove',
-                           'outbound', '--at=1')
-        except sh.ErrorReturnCode as e:
-            logger.error('Could not delete default rules for profile %s '
-                         '(assumed 2 inbound, 1 outbound)\n%s',
-                         self.profile_name, e)
-
-        # Call calicoctl to populate inbound rules
-        for rule in inbound_rules:
-            logger.info('applying inbound rule \n%s', rule)
-            try:
-                self.calicoctl('profile', self.profile_name, 'rule', 'add',
-                               'inbound', rule)
-            except sh.ErrorReturnCode as e:
-                logger.error('Could not apply inbound rule %s.\n%s', rule, e)
-
-        # Call calicoctl to populate outbound rules
-        for rule in outbound_rules:
-            logger.info('applying outbound rule \n%s', rule)
-            try:
-                self.calicoctl('profile', self.profile_name, 'rule', 'add',
-                               'outbound', rule)
-            except sh.ErrorReturnCode as e:
-                logger.error('Could not apply outbound rule %s.\n%s', rule, e)
-
-        logger.info('Finished applying rules.')
+            # Other namespaces use the DEFAULT_POLICY and any declared
+            # annotation based policy to create rules.
+            if DEFAULT_POLICY == 'ns_isolation':
+                # Isolate on namespace boundaries by default.
+                logger.debug("Default policy is namespace isolation")
+                inbound_rules = [allow_ns]
+                outbound_rules = [allow]
+            else:
+                # Allow all traffic by default.
+                logger.debug("Default policy is allow all")
+                inbound_rules = [allow]
+                outbound_rules = [allow]
+
+            logger.info("Checking for annotation based policy")
+            annotations = self._get_metadata(pod, "annotations")
+
+            # If policy annotations are defined, use them to generate rules.
+            if annotations and POLICY_ANNOTATION_KEY in annotations:
```

Comment:
Might be clearer if we make this part of the if/elif branch above? That way we only set the inbound_rules once, and don't rely on overwriting a previously-set value.